compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -670,7 +673,7 @@ impl<'tcx> UpvarSubsts<'tcx> {
     }
 
     #[inline]
-    pub fn tupled_upvars_ty(self) -> Ty<'tcx> {
+    pub fn tupled_upvars_ty(self) -> Result<Ty<'tcx>, ErrorReported> {
```

Comment:
Pre-existing to some extent, but I think this function merits a comment. Something like:

Returns a tuple type whose elements are the types of the upvars captured by this closure/generator. Returns `Err(ErrorReported)` if a type error prevented us from figuring out the types of the upvars for this closure.